### PR TITLE
Coin selection and PayAll

### PR DESCRIPTION
**Pull request by Jibz1**
- Coin selection update
- PayAll Sui whenever `sendMaxToken` is selected
- @mystie711 I can't style the input suffix

https://user-images.githubusercontent.com/8676844/218987462-f4dfaaae-7df2-4ab9-8e33-3fa9f3924349.mov

https://user-images.githubusercontent.com/8676844/218990042-f02924d0-5d74-4177-9d31-832eef472e38.mov

update
Screenshot 2023-02-15 at 10 28 02 AM

### CODE REVIEW

File: apps/wallet/src/ui/app/redux/slices/transactions/index.ts [hunk 1]
```diff
@@ -47,15 +53,25 @@ export const sendTokens = createAsyncThunk<
...
-                gasBudget
-            )
-        );
+        let response;
+        // Use payAllSui if sendMax is true and the token type is SUI
+        if (sendMax && tokenTypeArg === SUI_TYPE_ARG) {
+            response = await signer.payAllSui({
+                recipient: recipientAddress,
+                gasBudget: gasBudget,
+                inputCoins: coins.map((o) => o.fields.id.id),
```
[the remark this answers is not in the document]
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/redux/slices/transactions/index.ts:62]: `coins` is not only the SUI coins but all the other as well

File: apps/wallet/src/ui/app/components/active-coins-card/CoinItem.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    balance,
+}: {
+    coinType: string;
+    balance: bigint;
+}) {
+    const rpc = useRpc();
+
+    // Get coin metadata from the coin type
+    // Include icon url and name of the coin
+    const { data, isLoading } = useQuery(
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/active-coins-card/CoinItem.tsx:23]: Ideally this should go to it's own file I think. (We should probably add a folder in hooks and put there all the hooks related to api)

**Comment by mystie711** [comment]
Looking great overall @Jibz1. A few requests,

NEW: Let's remove the "available" text on the right,
image

BUG: The input field is tall when the form loads. Could we by default keep the standard size?
image

BUG: Estimated Gas Fees is using the incorrect info icon. Alignment should be centered vertically.
image

BUG: Valid address notification is using the wrong check icon. The visual treatment for positive feedback should be stretched full width to the input field.
image

PLEASE DOUBLE CHECK (updated Figma that was somehow not using components before): Font style for the address link on right
image

BUG?: Total Amount should be shown in absolute value and not rounded down
image

DOUBLE CHECK: Font syle for address
image

NEW: Let's center-align the SUI value on right vertically until we have the $ value
image

NEW: Let's also drop the top div (just like we do for the bottom one in the list)
image

🙏🏽

File: apps/wallet/src/ui/app/components/active-coins-card/CoinItem.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        ['get-coin-meta', coinType],
+        async () => {
+            return rpc.getCoinMetadata(coinType);
+        },
+        { enabled: !!coinType }
+    );
+
+    const [formatted, symbol] = useFormatCoin(balance, coinType);
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/active-coins-card/CoinItem.tsx:31]: This already fetches the metadata, it probably can expose it here as well.

File: apps/wallet/src/ui/app/components/coin-icon/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+});
+
+export interface CoinIconProps extends VariantProps<typeof imageStyle> {
+    coinType: string;
+}
+
+// fetch the coin metadata
+// show the coin icon
+// fallback to the coin name if the icon is not available
+// TODO: (jibz) use getCoinMetadata to get the coin metadata and use sui icons
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/coin-icon/index.tsx:31]: This seems simple to do, why punt?

**Comment by Jibz1** [comment]
Closing this PR for [#8353](https://github.com/MystenLabs/sui/pull/8353)

**Pull request closed without merging**